### ISSUE TITLE
OpenSSH is installed by default in Fedora minimal template (since the last upgrade)

### ISSUE THREAD

**Issue by TNTBOMBOM** [QubesOS/qubes-issues#9045]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)

OpenSSH, particularly the server component, shouldn't be installed on a user's desktop by default. It's a specialized package that primarily appeals to users who wish to host services or log into external servers, a group that represents a minority of users, without question. Moreover, installing this package by default could pose a security risk under unfortunate circumstances.

### The solution you'd like

It shouldn't be included by default; don't rely on it being there.

### The value to a user, and who that user might be

The DVM will primarily be based on Fedora-minimal/Fedora-full, and consequently, sys-net/firewall and similar components will also be based on it. Including this package in such a configuration is not advisable.

### Screenshots

![openssh](https://github.com/QubesOS/qubes-issues/assets/11895339/6a886e2b-95b0-4ad4-8b72-48d81c3fa9d7)

![openssh2](https://github.com/QubesOS/qubes-issues/assets/11895339/ff9e96a4-6532-43bb-bcc2-c5a7f1aa7c6a)

* But this behavior not spotted on debian-minimal:

![openssh3](https://github.com/QubesOS/qubes-issues/assets/11895339/e4932aba-d430-4f73-84ad-e49c58d5ccb8)

**Comment by xaki23** [comment]
fwiw ... openssh-server is considered a "mandatory package" for a fedora "smallest possible installation".
see ... dnf group info core

**Comment by TNTBOMBOM** [comment]
> fwiw ... openssh-server is considered a "mandatory package" for a fedora "smallest possible installation". see ... dnf group info core

The input seems to be overlooking a crucial point, as indicated by the ticket, The entire OpenSSH situation emerged just two days ago; it wasn't included by default before, and it shouldn't be now. (just because you see it categorized as a mandatory package after it got installed, doesnt make it mandatory)

**Comment by xaki23** [comment]
>  (just because you see it categorized as a mandatory package after it got installed, doesnt make it mandatory)

there actually was a discussion about this some weeks ago on the matrix chat. which had certain members of the qubes team argue that all packages from the core-mandatory group should be included. so this might be an intentional change. 
(i will ping them and ask for input on this)

**Comment by rustybird** [comment]
Duplicate of #9043

**Comment by DemiMarie** [comment]
Installing the OpenSSH client _might_ make sense.  Installing the OpenSSH _server_ makes _zero_ sense, not least because OpenSSH’s server needs special configuration (ensuring that the host keys are on the private volume) before it can be securely used in Qubes OS.

**Comment by bi0shacker001** [comment]
> Installing the OpenSSH client _might_ make sense. Installing the OpenSSH _server_ makes _zero_ sense, not least because OpenSSH’s server needs special configuration (ensuring that the host keys are on the private volume) before it can be securely used in Qubes OS.

Is there a solution that's currently used for the non-minimal templates for this? 
This is only being brought up for minimal, but the package exists in the other f38 templates too, so this may be a bigger issue if that's the case. If it's resolved in those, then that's likely a suitable mitigation here too.

**Comment by DemiMarie** [comment]
I don’t think this update pulls in the OpenSSH server.  It pulls in the OpenSSH _client_, but that has no security ramifications.

**Comment by bi0shacker001** [comment]
> I don’t think this update pulls in the OpenSSH server. It pulls in the OpenSSH _client_, but that has no security ramifications.

It looks like it pulls them both in. openssh is pulled in as a dependency (which I think is weird, but maybe that's just me, haha)

**Comment by DemiMarie** [comment]
> > I don’t think this update pulls in the OpenSSH server. It pulls in the OpenSSH _client_, but that has no security ramifications.
> 
> It looks like it pulls them both in. openssh is pulled in as a dependency (which I think is weird, but maybe that's just me, haha)

`openssh` just provides `ssh-keygen` and `ssh-keysign`.  `openssh-client` is the client.  The server is in `openssh-server`, which should not be pulled in.

I think `openssh-client` is pulled in by `gnome-keyring`, which is needed to ensure that Flatpak applications do not break when one switches templates.  Some applications are buggy enough to cause irreversable data loss if there is no secret portal implementation.

**Comment by TNTBOMBOM** [comment]
> openssh just provides ssh-keygen and ssh-keysign

I see, if so then both packages (openssh-client, openssh) are considered safe. (though not that much wanted in a minimal template)..

So i will leave this ticket open if there is an interest to get rid of openssh entirely (client or not). Otherwise feel free to close the ticket.

**Comment by marmarek** [comment]
> I think `openssh-client` is pulled in by `gnome-keyring`, which is needed to ensure that Flatpak applications

But minimal template doesn't need to have functioning flatpak. What pulls in `xdg-desktop-portal` in the minimal template? If that's really "core" package in Fedora, then I think this is non-issue. But if not, we can try to get it not installed in the minimal template.

**Comment by DemiMarie** [comment]
> > I think `openssh-client` is pulled in by `gnome-keyring`, which is needed to ensure that Flatpak applications
> 
> But minimal template doesn't need to have functioning flatpak. What pulls in `xdg-desktop-portal` in the minimal template? If that's really "core" package in Fedora, then I think this is non-issue. But if not, we can try to get it not installed in the minimal template.

`xdg-desktop-portal` is used by many GUI frameworks for e.g. file open and save dialogs.  This might be a good question to ask Fedora.

**Comment by ben-grande** [comment]
I have no idea what pulls `openss-clients`. I used `rpm -q --what(requires|recommends|suggests) openssh-clients`, but nothing appeared. OpenSSH server is not installed on fedora-43-minimal.

> fwiw ... openssh-server is considered a "mandatory package" for a fedora "smallest possible installation". see ... dnf group info core

As said above, it appears on "mandatory package", both `openssh-clients` and `openssh-server`, but `openssh-server` is not pulled.
